core/src/subgraph/instance_manager.rs

```diff
@@ -721,6 +725,24 @@ where
 
             match res {
                 Ok(needs_restart) => {
+                    if should_try_update_sync_status(
```

Comment:
I'd inline `should_try_update_sync_status` here, `!synced` is such a simple condition that it's not worth testing for imo. This way you don't need the closure anymore. But up to you.

Reply:
Removed!